### PR TITLE
Allow the use of 'true', 't', 'false', 'f', '1', '0' in addition to '*' when populating a field in REST API

### LINKED ISSUE #13945: Image relation has an unexpected 'related' object that contains duplicate of the root document.
I'm getting a weird response with image population:

When I request an article collection with http://localhost:1337/api/Articles?populate[image][populate]=*

I get back a normal-looking response, but the Image node has a nested 'related' node that includes all my data for the parent Article object!

So, it kind of repeats the Article data in an Image-node property called 'related'.  It means I can't stringify it, because it's a circular reference.  I've checked my Strapi content type and it's just a normal 'Image /Media type' and the API code doesn't create an Image type explicitly, because it's built-in.  Something odd is going on here, but I can't figure out what...

I did upgrade from V3 to V4, is it possibly something to do with that?  I followed guides and used CodeMods back in June to migrate.  I went from v3 to 4.1 to 4.2 to 4.3.beta-1, 4.3.beta-2 and now 4.3.

- Node.js version: 16.14.2
- NPM version: 8.7.0
- Strapi version: 4.3.0
- Database: Postgres
- Operating system: win 10 pro 64

### Describe the bug

This image object (below) appears in an article with ID 2016.  The same ID as the article that appears in the 'related' field of the image object.
I would not expect to find related articles in an image object and I don't know how it got there.  I can see how this might happen, but is this a config option or a bug?  I can't find anything at all the sheds light on it.

```ts
                "image": {
                    "data": {
                        "id": 1381,
                        "attributes": {
                            "name": "/media/70387/style-625px.jpg",
                            "alternativeText": null,
                            "caption": null,
                            "width": 625,
                            "height": 327,
                            "formats": {
                                "small": {
...
                                },
                                "thumbnail": {
...
                                }
                            },
                            "hash": "style_625px_ba1251f301",
                            "ext": ".jpg",
                            "mime": "image/jpeg",
                            "size": 23.6,
                            "url": "https://127.0.0.1:10000/devstoreaccount1/test-strapi-uploads/assets/bhot.jpg",
                            "previewUrl": null,
                            "provider": "azure-storage",
                            "provider_metadata": null,
                            "createdAt": "2022-01-11T11:25:55.250Z",
                            "updatedAt": "2022-01-11T11:25:55.250Z",
                            "related": {
                                "data": [
                                    {
                                        "id": 2016,
                                        "attributes": {
                                            "__type": "api::article.article",
                                            "title": "A Brief History of Time",
                                            "description": "A Brief History of Time ",
                                            "content": "<p>[my article]</p>
                                            "slug": "a-brief-history-of-time",
                                            "updated": null,
                                            "authoredDate": null,
                                            "createdAt": null,
                                            "updatedAt": null,
                                            "publishedAt": "2022-01-11T11:25:34.254Z"
                                        }
                                    }
                                ]
                            }
                        }
                    }
                },

```

### PR DESCRIPTION
**Pull request by petersg83**
### What does it do?

To populate a field, we can now do:
```js
populate: {
  image: 'true', // same with 't', 'false', 'f', '1', '0' and '*'
}
```

### Why is it needed?

Before we could only do `image: '*'` which may not be explicit about what it does :
- does it populate everything in the image like `{ populate: '*' }` would do
- does it populate only with the scalar attributes (it's this one)

Using `true` may be more explicit that it doesn't populate relation under `image`

### How to test it?

A test was added

### Related issue(s)/PR(s)

fix #13945

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
I'm gonna be that guy, but can we also make sure we add this to the documentation in some way? maybe just a callout?

**Comment by petersg83** [comment]
I forgot to put the label but yes, it's on my to do to update the doc too! So you're not that guy :p

**Review by Marc-Roig — APPROVED** [review]

**Comment by petersg83** [comment]
Documentation PR : https://github.com/strapi/documentation/pull/1071

File: packages/core/utils/lib/convert-query-params.js [hunk 1]
```diff
@@ -215,6 +215,10 @@ const convertNestedPopulate = (subPopulate, schema) => {
     return true;
   }
 
+  if (_.isString(subPopulate)) {
+    return parseType({ type: 'boolean', value: subPopulate, forceCast: true });
```
[the remark this answers is not in the document]
Comment (petersg83) [reply packages/core/utils/lib/convert-query-params.js:219]: It allow extra values, it does `return Boolean(value)`. So it will be equivalent to true. I've added this option in order to avoid 500 errors because it's a regular error that is thrown. We can try catch somewhere to transform the error but I didn't think it was worth it. wdyt?

`false` will normally not populate the field (the function already can return false line 222). I remember allowing having false to not populate but I'm not sure if it was merged in main or not. I'll add a test :)

[the remark this answers is not in the document]
Comment (petersg83) [reply packages/core/utils/lib/convert-query-params.js:219]: Indeed `populate: false` was not supported in main. I added it in the PR with a test!

File: packages/core/strapi/tests/api/populate/filtering/index.test.e2e.js [hunk 2]
```diff
@@ -300,5 +300,19 @@ describe('Populate filters', () => {
 
       expect(body.data[0].attributes.third.data[0].attributes.fooRef).toBeUndefined();
     });
+
+    test("Don't populate with object and 'f'", async () => {
+      const qs = {
+        populate: {
+          third: 'f',
```
Comment (alexandrebodin) [inline packages/core/strapi/tests/api/populate/filtering/index.test.e2e.js:307]: I'm not sure what I think about this. IMO it doesn't make sense to make this a possibility as our populate algorithm only works by picking what you want not omitting what you don't want 🤔 

It would make sense if we could say populate everything but `third` but that's not the case.

Comment (petersg83) [reply packages/core/strapi/tests/api/populate/filtering/index.test.e2e.js:307]: I understand. I see it as "If I can put `true` than I can also put `false`". It's weird to put false and to see it populated.
I also imagine it of how it can be used programmatically: `shouldPopulate ? true : false` (or just `shouldPopulate`).

**Comment by gu-stav** [comment]
My 2cts: I'm not very happy with the syntax addition tbh and I think it makes it more ambigious. Currently the string that is being passed looks and behaves like a glob. If we start allowing truthy values, it would open questions such as:

- does `true` mean `*`?
- does `true` mean `**` (once added)?
- does `0` mean 0 levels? does `1` mean 1 level?

Therefore I think we shouldn't start mixing globbing and boolean syntax.

**Comment by petersg83** [comment]
But it's not a glob value. `*` doesn't say "populate all the fields bellow", its says "yes, populate this field". It doesn't make sense for me to put a `*` for something that doesn't mean "populate all". That's why I found it confusing and `true` makes more sense to me.
If we want the `fieldName: value` to be a shorcut for the populate param, then the `*` should populate all fields bellow imo. And in this case it would make sense to have the coexistance of `true` (0 level bellow), `*` (1 level bellow), `*.*` etc.

**Comment by gu-stav** [comment]
I agree that as long as there is only `*` it does not make too much sense, but as soon as there will be `**` or `*.*` it does. 

- `*` current level
- `**` current plus all other levels
- `*.*` current level + 1

I'm not sure if is planned that way, but to me that would make sense and be less ambigious than having this syntax + boolean values.

**Comment by Convly** [comment]
Hey, here are my 2 cents: I agree with Gustav here. I don't find the addition of 't', 'f', 0, and 1 very useful. I would prefer if we stick with a single syntax (glob?) that we extend iteratively (feature-wise) instead of adding aliases for the same behavior. I find it very confusing.

File: packages/core/utils/lib/convert-query-params.js [hunk 3]
```diff
@@ -215,6 +238,10 @@ const convertNestedPopulate = (subPopulate, schema) => {
     return true;
```
Comment (alexandrebodin) [inline packages/core/utils/lib/convert-query-params.js:238]: Can we remove this if we just do a truthy check?

File: packages/core/utils/lib/convert-query-params.js [hunk 4]
```diff
@@ -203,9 +220,15 @@ const convertPopulateObject = (populate, schema) => {
       return acc;
     }
 
+    const populateObject = convertNestedPopulate(subPopulate, targetSchema);
+
+    if (!populateObject) {
```
Comment (alexandrebodin) [inline packages/core/utils/lib/convert-query-params.js:225]: Should we do a strict false check here ?

File: packages/core/utils/lib/convert-query-params.js [hunk 5]
```diff
@@ -174,7 +183,15 @@ const convertPopulateObject = (populate, schema) => {
     if (attribute.type === 'dynamiczone') {
       const populates = attribute.components
         .map((uid) => strapi.getModel(uid))
-        .map((schema) => convertNestedPopulate(subPopulate, schema));
+        .map((schema) => {
+          const populate = convertNestedPopulate(subPopulate, schema);
+          return isBoolean(populate) && populate ? {} : populate;
```
[the remark this answers is not in the document]
Comment (petersg83) [reply packages/core/utils/lib/convert-query-params.js:188]: It's because we are merging the populate later on and a boolean value would mess with it.

File: packages/core/utils/lib/convert-query-params.js (hunk 4 above)
Comment (petersg83) [reply packages/core/utils/lib/convert-query-params.js:225]: Yes it's the same. It should be either an `object` or `true` or `false`.

File: packages/core/utils/lib/convert-query-params.js (hunk 3 above)
Comment (petersg83) [reply packages/core/utils/lib/convert-query-params.js:238]: I'm not sure. We want to convert the populate param to DB params, so `true` is better than `*` no?

Comment (alexandrebodin) [reply packages/core/utils/lib/convert-query-params.js:238]: Let me rephrase '*' is a string so it will be converted to true in the isString just below. making this if unnecessary :)

File: packages/core/utils/lib/convert-query-params.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (alexandrebodin) [reply packages/core/utils/lib/convert-query-params.js:188]: we where already returning boolean values in some cases before though 🤔 was it a bug before ?

[the remark this answers is not in the document]
Comment (petersg83) [reply packages/core/utils/lib/convert-query-params.js:188]: Yes it was bug that never appeared I think :P

File: packages/core/utils/lib/convert-query-params.js (hunk 3 above)
Comment (petersg83) [reply packages/core/utils/lib/convert-query-params.js:238]: oh ok yes :p Let's remove it

File: packages/core/utils/lib/convert-query-params.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (alexandrebodin) [reply packages/core/utils/lib/convert-query-params.js:188]: then I would suggest

```suggestion
          return convertNestedPopulate(subPopulate, schema) || {}
```

[the remark this answers is not in the document]
Comment (petersg83) [reply packages/core/utils/lib/convert-query-params.js:188]: I don't think it would work as I need to remove the populate if it's `false` (let's say there is only one populate that is `false`, I want the `populates` object to be empty) and I need it to be an empty object if it's `true`.

I agree it not a really pretty code. I can do 2 maps instead of one maybe?

**Review by alexandrebodin — APPROVED** [review]
